openpype/hosts/maya/plugins/load/load_reference.py

```diff
@@ -51,7 +51,9 @@ def process_reference(self, context, name, namespace, options):
 
         with maintained_selection():
             cmds.loadPlugin("AbcImport.mll", quiet=True)
-            nodes = cmds.file(self.fname,
+            file_url = self.prepare_root_value(self.fname,
+                                               context["project"]["code"])
+            nodes = cmds.file(file_url,
```

Comment:
Somehow it looks to me like this particular change - if applicable to **all** references should actually happen in the `ReferenceLoader` class in `load` and potentially just do `self.fname = self.prepare_root_value(self.fname)`.

Just so the logic is encapsulated to that area only?

We should be able to make that change ONLY in the `ReferenceLoader` class and have that work for all implementations already, right?

Reply:
Imho no, each inheriting class of `ReferenceLoader` could calculate file url differently (or at least some existing loaders do). Plus we sometimes need it in update logic.